### PR TITLE
feat(gateway): allow specifying a cloud environment using the version lock header

### PR DESCRIPTION
**Pull request by trieloff**

### CODE REVIEW

File: test/openwhisk.integration.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ describe('OpenWhisk Integration Test', () => {
     const out = builder.cfg._logger.output;
     const { auth, namespace } = builder._deployers.wsk._cfg;
     assert.ok(out.indexOf(`ok: 200
-{"url":"https://adobeioruntime.net/api/v1/web/${namespace}/simple-package/simple-name@1.45.0/foo","file":"Hello, world.\\n"}`) > 0, out);
+{"url":"/api/v1/web/${namespace}/simple-package/simple-name@1.45.0/foo","file":"Hello, world.\\n"}`) > 0, out);
```
Comment (tripodsan) [inline test/openwhisk.integration.js:66]: hmm... this would be the wrong output

**Review by tripodsan — CHANGES_REQUESTED** [review]
don't change the output

File: test/openwhisk.integration.js (hunk 1 above)
Comment (trieloff) [reply test/openwhisk.integration.js:66]: yep

File: test/openwhisk.integration.js [hunk 2]
```diff
@@ -63,7 +63,7 @@ describe('OpenWhisk Integration Test', () => {
     const out = builder.cfg._logger.output;
     const { auth, namespace } = builder._deployers.wsk._cfg;
     assert.ok(out.indexOf(`ok: 200
-{"url":"/api/v1/web/${namespace}/simple-package/simple-name@1.45.0/foo","file":"Hello, world.\\n"}`) > 0, out);
+{"url":"https://azure.adobe-runtime.com/api/v1/web/${namespace}/simple-package/simple-name@1.45.0/foo","file":"Hello, world.\\n"}`) > 0, out);
```
Comment (tripodsan) [inline test/openwhisk.integration.js:66]: ???

Comment (trieloff) [reply test/openwhisk.integration.js:66]: Runtime changed the output, and now tests are failing.

Comment (tripodsan) [reply test/openwhisk.integration.js:66]: I think the output should remain the same, i.e. use the raw url as defined by the environment

Comment (tripodsan) [reply test/openwhisk.integration.js:66]: > Runtime changed the output, and now tests are failing.

oh!

**Review by tripodsan — APPROVED** [review]

File: test/openwhisk.integration.js (hunk 2 above)
Comment (trieloff) [reply test/openwhisk.integration.js:66]: "oh!" is way more polite than what I thought.

**Review by tripodsan — APPROVED** [review]

**Pull request merged**